packages/server-wallet/src/wallet/wallet.ts

```diff
@@ -33,32 +38,73 @@ export class Wallet extends EventEmitter<ObjectiveProposed> {
    */
   public static async create(
     engine: Engine,
+    chainService: ChainServiceInterface,
     messageServiceFactory: MessageServiceFactory,
     retryOptions: Partial<RetryOptions> = DEFAULTS
   ): Promise<Wallet> {
-    return new Wallet(messageServiceFactory, engine, {...DEFAULTS, ...retryOptions});
+    await chainService.checkChainId(engine.engineConfig.networkConfiguration.chainNetworkID);
+    const wallet = new Wallet(messageServiceFactory, engine, chainService, {
+      ...DEFAULTS,
+      ...retryOptions,
+    });
+    await wallet.registerExistingChannelsWithChainService();
+    return wallet;
   }
 
   private _messageService: MessageServiceInterface;
+  private _destroyed = false;
 
   private constructor(
     messageServiceFactory: MessageServiceFactory,
     private _engine: Engine,
+    private _chainService: ChainServiceInterface,
     private _retryOptions: RetryOptions
   ) {
     super();
+
     const handler: MessageHandler = async message => {
-      const {outbox, newObjectives} = await this._engine.pushMessage(message.data);
+      const {outbox, newObjectives, channelResults} = await this._engine.pushMessage(message.data);
       // Receiving messages from other participants may have resulted in new proposed objectives
       for (const o of newObjectives) {
         this.emit('ObjectiveProposed', o);
+
+        // If this is a new open channel objective that means there is a new channel to be monitored in the chain service
+        if (o.type === 'OpenChannel') {
+          const listener = createChainListener(
+            this._engine,
+            this._engine.store,
+            this.messageService
+          );
+          const assetHolders = _.uniq(
+            _.flatten(channelResults.map(cr => cr.allocations.map(a => a.assetHolderAddress))).map(
+              makeAddress
+            )
+          );
+          this._chainService.registerChannel(o.data.targetChannelId, assetHolders, listener);
+        }
       }
       await this.messageService.send(getMessages(outbox));
     };
 
     this._messageService = messageServiceFactory(handler);
   }
-
+  /**
+   * Pulls and stores the ForceMoveApp definition bytecode at the supplied blockchain address.
+   *
+   * @remarks
+   * Storing the bytecode is necessary for the engine to verify ForceMoveApp transitions.
+   *
+   * @param  appDefinition - An ethereum address where ForceMoveApp rules are deployed.
+   * @returns A promise that resolves when the bytecode has been successfully stored.
+   */
+  public async registerAppDefinition(appDefinition: string): Promise<void> {
```

Comment:
Doesn't seem like this function is used anywhere at the moment

Reply:
This was moved from the `engine`. I think it could be useful as a external API method so I'd like to keep it there for the moment.